src/rdkafka_buf.h

```diff
@@ -125,7 +125,7 @@ rd_tmpabuf_write0 (const char *func, int line,
 		   rd_tmpabuf_t *tab, const void *buf, size_t size) {
 	void *ptr = rd_tmpabuf_alloc0(func, line, tab, size);
 
-	if (ptr)
```

Comment:
I'd argue that `if (ptr && size)` is more correct, to catch bad callers.
If that messes up the compiler optimization, we could do:
```
 if (likely(ptr && buf)) {
       rd_dassert(size > 0);  <-- that's a ENABLE_DEVEL-only assert
       memcpy..
}
```

Reply:
I did a quick audit of other memcpy and friends usage, but couldn't see anything wrong. I'm not familiar enough with the code base though.